src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -134,7 +134,7 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         }
         Set<Medication> updatedMedications = editPersonDescriptor.getMedications()
                 .orElse(personToEdit.getMedications());
-        List<PastAppointment> pastAppointments = editPersonDescriptor.getPastAppointments();
+        List<PastAppointment> pastAppointments = personToEdit.getPastAppointments();
```

Comment:
Good catch on this error! Fixed cause for editing of a person details to delete all past appointments.